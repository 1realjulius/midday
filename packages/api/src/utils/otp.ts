export const generateOtp = (limit = 6) => {
  const digits = "0123456789";
  let OTP = "";

  for (let i = 0; i < limit; i++) {
    OTP += digits[Math.floor(Math.random() * 10)];
  }

  return OTP;
};

export interface SmsOptions {
  to: string;
  text: string;
}

export const sendSMS = async ({ to, text }: SmsOptions) => {
  try {
    const response = await fetch("https://rest.nexmo.com/sms/json", {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        from: "Lasagne",
        to,
        text,
        api_key: process.env.VONAGE_API_KEY!,
        api_secret: process.env.VONAGE_API_SECRET!,
      }),
    });

    return response.json();
  } catch (error) {
    console.log(error);
  }
};
