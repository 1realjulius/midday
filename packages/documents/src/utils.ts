export const findValue = (entities, type: string) => {
  const found = entities.find((entry) => entry.type === type);
  return found?.normalizedValue?.text || found?.mentionText;
};
