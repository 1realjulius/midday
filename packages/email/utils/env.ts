export function getEnvironmentUrl() {
  if (process.env.NODE_ENV === "development") {
    return "http://localhost:3000/email";
  }

  return "https://midday.ai/email";
}
