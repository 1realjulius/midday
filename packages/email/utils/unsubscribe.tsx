import { SignJWT, jwtVerify } from "jose";

const secretKey = new TextEncoder().encode(process.env.UNSUBSCRIBE_JWT_SECRET!);

type GenerateUnsubscribeLinkParams = {
  id: string;
  type: "onboarding";
};

export async function generateUnsubscribeLink({
  id,
  type,
}: GenerateUnsubscribeLinkParams) {
  const token = await new SignJWT({
    id,
    type,
  })
    .setProtectedHeader({ alg: "HS256" })
    .setIssuedAt()
    .setExpirationTime("1 day")
    .sign(secretKey);

  const url = new URL(`/unsubscribe/${token}`, "https://app.midday.ai");

  url.searchParams.append("type", type);

  return url.toString();
}

export async function verifyUnsubscribeLink(token: string) {
  try {
    const { payload } = await jwtVerify(token, secretKey, {
      issuer: process.env.UNSUBSCRIBE_JWT_ISSUER,
    });

    return payload;
  } catch (e) {
    console.log("Token is invalid");
  }
}
