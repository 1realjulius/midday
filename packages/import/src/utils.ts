import crypto from "node:crypto";
import type { Transaction } from "@midday/providers/src/types";
import { capitalCase } from "change-case";
import { parse } from "csv-parse/sync";
import { formatISO } from "date-fns";
import { z } from "zod";
import type { ExtractedTransaction, FindIndexesByKeyParams } from "./types";

function generateId(value: string) {
  const hash = crypto.createHash("sha256");
  hash.update(value);

  return hash.digest("hex");
}

type TransformTransaction = Transaction & { teamId: string };

export function transformAmount(amount: string | number) {
  if (typeof amount === "string") {
    return +amount.replace(/\s/g, "").replace(",", ".");
  }

  return amount;
}

export function transform(transaction: TransformTransaction) {
  const internalId = generateId(
    `${transaction.date}-${transaction.name}-${transaction.amount}`
  );
  return {
    internal_id: internalId,
    team_id: transaction.teamId,
    status: "posted",
    date: formatISO(transaction.date, { representation: "date" }),
    amount: transformAmount(transaction.amount),
    name: transaction?.description && capitalCase(transaction.description),
    manual: true,
  };
}

export function parseCsv(input: string) {
  return parse(input, {
    delimiter: [",", ";", "."],
    skip_empty_lines: true,
    skip_records_with_empty_values: true,
    skip_records_with_error: true,
    trim: true,
    relax_column_count: true,
  });
}

export const transactionSchema = z.object({
  amount: z.string(),
  description: z.string(),
  date: z.coerce.date(),
});

export const filterTransactions = (row: ExtractedTransaction) => {
  const parsed = transactionSchema.safeParse(row);

  if (parsed.success) {
    return true;
  }

  return false;
};

export const findIndexesByKey = ({
  raw,
  input,
  key,
  parse = (value) => value.toString().trim(),
}: FindIndexesByKeyParams) => {
  const values = input.map((v) => parse(v[key]));

  const matchIndices = raw.map((r) =>
    r.findIndex((v) => values.includes(parse(v)))
  );

  return Math.max(...matchIndices);
};
