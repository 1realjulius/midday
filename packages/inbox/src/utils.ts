export function getInboxIdFromEmail(email: string) {
  return email.split("@").at(0);
}

export const findDocumentValue = (entities, type) => {
  const found = entities.find((entry) => entry.type === type);
  return found?.normalizedValue?.text || found?.mentionText;
};
