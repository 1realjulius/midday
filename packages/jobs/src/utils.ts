import { capitalCase } from "change-case";

export const mapTransactionMethod = (method: string) => {
  switch (method) {
    case "Payment":
    case "Bankgiro payment":
    case "Incoming foreign payment":
      return "payment";
    case "Card purchase":
    case "Card foreign purchase":
      return "card_purchase";
    case "Card ATM":
      return "card_atm";
    case "Transfer":
      return "transfer";
    default:
      return "other";
  }
};

export async function processPromisesBatch(
  items: Array<any>,
  limit: number,
  fn: (item: any) => Promise<any>
): Promise<any> {
  let results = [];
  for (let start = 0; start < items.length; start += limit) {
    const end = start + limit > items.length ? items.length : start + limit;

    const slicedResults = await Promise.all(items.slice(start, end).map(fn));

    results = [...results, ...slicedResults];
  }

  return results;
}

export const transformTransactions = (transactions, { teamId, accountId }) =>
  // We want to insert transactions in reversed order so the incremental id in supabase is correct
  transactions?.reverse().map((data) => ({
    transaction_id: data.transactionId,
    reference: data.entryReference,
    booking_date: data.bookingDate,
    date: data.valueDate,
    name: capitalCase(data.additionalInformation),
    original: data.additionalInformation,
    method: mapTransactionMethod(data.proprietaryBankTransactionCode),
    internal_id: data.internalTransactionId,
    amount: data.transactionAmount.amount,
    currency: data.transactionAmount.currency,
    bank_account_id: accountId,
    category: data.transactionAmount.amount > 0 ? "income" : null,
    team_id: teamId,
  }));
