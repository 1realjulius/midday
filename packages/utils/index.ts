export function stripSpecialCharacters(inputString: string) {
  // Use a regular expression to replace all non-alphanumeric characters except hyphen, space, dot,and parentheses with an empty string
  return inputString.replace(/[^a-zA-Z0-9\s.()-]/g, "");
}

export function shuffle(array: any) {
  for (let i = array.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [array[i], array[j]] = [array[j], array[i]];
  }
  return array;
}
