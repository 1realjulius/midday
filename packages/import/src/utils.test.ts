import { expect, test } from "bun:test";
import { csvTransformed } from ".";
import { parseCsv } from "./utils";

const csv1 = `Bokförd,Valutadatum,Text,Typ,Insättningar/uttag,Bokfört saldo
2024-03-08,2024-03-08,ELLEVIO AB (PUBL),Betalning (bg/pg),-168.00,1942493.52
2024-03-08,2024-03-08,ELLEVIO AB (PUBL),Betalning (bg/pg),-165.00,1942661.52
2024-03-06,2024-03-06,BANKTJÄNSTER,Annan,-136.00,1942826.52
2024-03-04,2024-03-04,KOSTNAD,Annan,-31.85,1942962.52
2024-03-04,2024-03-04,H04510236587,Internationell betalning,230400.00,1942994.37
2024-03-04,2024-03-03,AWS AMAZON C/24-03-02,Kortköp,-2797.17,1712594.37
2024-03-04,2024-03-02,GOOGLE  YOUT/24-03-01,Kortköp,-119.00,1715391.54
2024-03-04,2024-03-02,DUBLIN      /24-03-01,Kortköp,-119.40,1715510.54
2024-03-04,2024-03-02,DUBLIN      /24-03-01,Kortköp,-179.11,1715629.94
2024-03-04,2024-03-02,GSUITE WDLIN/24-03-01,Kortköp,-29.00,1715809.05
2024-03-04,2024-03-02,GOOGLE CLOUD/24-03-01,Kortköp,-39.85,1715838.05
2024-03-01,2024-03-01,AMSTERDAM   /24-03-01,Kortköp,-153.10,1715877.90`;

const csv2 = `"Kontonummer";"Kontonamn";"";"Saldo";"Tillgängligt belopp"
"90248745117";"Rörelsekonto";"";"195 460,31";"195 460,31"

"Bokföringsdatum";"Transaktionsdatum";"Transaktionstyp";"Meddelande";"Belopp"
"2024-03-08";"2024-03-08";"Kortköp";"APL*APPLE SEARCH ADS,800-275-2273,US";"-387,27"
"2024-03-08";"2024-03-08";"Betalning";"SKATTEVERKET";"-44 134,00"
"2024-03-08";"2024-03-08";"Överföring";"TILL SKV";"44 134,00"
"2024-03-08";"2024-03-08";"Betalning";"hallon / Billogram";"-485,00"
"2024-03-07";"2024-03-06";"Kortköp";"WEBHALLENE";"-8 007,00"
"2024-03-07";"2024-03-07";"Utlandsbetalning";"ANK Utland Normal";"2 590,96"
"2024-03-06";"2024-03-06";"Överföring";"Skatt moms pwm";"-48 000,00"
"2024-03-06";"2024-03-06";"BG-insättning";"52762176";"137 500,00"
"2024-03-06";"2024-03-06";"Utlandsbetalning";"ANK Utland Normal";"1 923,95"
"2024-03-05";"2024-03-05";"Kortköp";"FIGMA MONTHLY RENEWAL,SAN FRANCISCO,US";"-316,17"
"2024-03-05";"2024-03-04";"Kortköp";"PAYPAL *APPERS GMBH,35314369001,AT";"-806,18"
"2024-03-02";"2024-03-01";"Kortköp";"GOOGLE*CLOUD 5NHT7Z,CC GOOGLE.COM,IE";"-1,22"
"2024-03-02";"2024-03-01";"Kortköp";"GSUITE,Dublin,IE";"-100,00"`;

const csv3 = `Bokförd Valutadatum Text Typ Insättningar/uttag "Bokfört saldo"
2024-05-17 2024-05-17 "SAN FRANCISC/24-05-16" Kortköp -714,26 70001,16
2024-05-15 2024-05-15 SKATTEVERKET "Betalning (bg/pg)" -67749,00 70715,42
2024-05-08 2024-05-08 "VOLKSWAGEN FINANS SVERIG" "Betalning (bg/pg)" -6615,00 138464,42
2024-05-07 2024-05-07 "PREMIE FÖRS." Annan -5000,00 145079,42
2024-05-06 2024-05-06 "TWITTER PAID/24-05-05" Kortköp -166,00 150079,42
2024-05-06 2024-05-06 BANKTJÄNSTER Annan -264,25 150245,42`;

const csv4 = `Bokförd.Valutadatum.Text.Typ.Insättningar/uttag.Bokfört saldo
2024-05-17.2024-05-17.SAN FRANCISC/24-05-16.Kortköp.-714,26.70001,16
2024-05-15.2024-05-15.SKATTEVERKET.Betalning (bg/pg).-67749,00.70715,42
2024-05-08.2024-05-08.VOLKSWAGEN FINANS SVERIG.Betalning (bg/pg).-6615,00.138464,42
2024-05-07.2024-05-07."PREMIE FÖRS.".Annan.-5000,00.145079,42
2024-05-06.2024-05-06.TWITTER PAID/24-05-05.Kortköp.-166,00.150079,42
2024-05-06.2024-05-06.BANKTJÄNSTER.Annan.-264,25.150245,42
2024-05-06.2024-05-06.KOSTNAD.Annan.-31,85.150509,67
2024-05-06.2024-05-06.H06513143882.Internationell betalning.150000,00.150541,52
2024-05-03.2024-05-03.DUBLIN      /24-05-02.Kortköp.-99,11.541,52`;

test("Parse from comma delimiter", () => {
  expect(parseCsv(csv1)).toMatchSnapshot();
});

test("Parse from colon delimiter", () => {
  expect(parseCsv(csv2)).toMatchSnapshot();
});

test("Parse from space delimiter", () => {
  expect(parseCsv(csv3)).toMatchSnapshot();
});

test("Parse from dot delimiter", () => {
  expect(parseCsv(csv4)).toMatchSnapshot();
});

test("Transform transactions", () => {
  expect(
    csvTransformed({
      teamId: "123",
      extracted: [
        {
          date: "2024-03-03",
          description: "AWS AMAZON C/24-03-02",
          amount: -2797.17,
        },
        {
          date: "2024-03-02",
          description: "GOOGLE YOUT/24-03-01",
          amount: -119,
        },
        {
          date: "2024-03-04",
          description: "DUBLIN /24-03-01",
          amount: -119.4,
        },
        {
          date: "2024-03-02",
          description: "DUBLIN /24-03-01",
          amount: -179.11,
        },
        {
          date: "2024-03-02",
          description: "GSUITE WDLIN/24-03-01",
          amount: -29,
        },
        {
          date: "2024-03-02",
          description: "GOOGLE CLOUD/24-03-01",
          amount: -39.85,
        },
        {
          date: "2024-03-01",
          description: "AMSTERDAM /24-03-01",
          amount: -153.1,
        },
        {
          date: "2024-02-29",
          description: "TELE2 SV AB FÖRETAG ATTN",
          amount: -971,
        },
        {
          date: "2024-02-27",
          description: "SAN FRANCISC/24-02-26",
          amount: -42.47,
        },
        {
          date: "2024-02-27",
          description: "PARIS /24-02-26",
          amount: -1061.65,
        },
      ],
      raw: csv1,
    })
  ).toMatchSnapshot();
});

test("Transform transactions", () => {
  expect(
    csvTransformed({
      teamId: "123",
      extracted: [
        { date: "2024-03-08", description: "Överföring", amount: 44134 },
        { date: "2024-03-08", description: "Betalning", amount: -485 },
        { date: "2024-03-07", description: "Kortköp", amount: -8007 },
        { date: "2024-03-06", description: "Överföring", amount: -48000 },
        { date: "2024-03-05", description: "Kortköp", amount: -316.17 },
        { date: "2024-03-05", description: "Kortköp", amount: -806.18 },
        { date: "2024-03-02", description: "Kortköp", amount: -1.22 },
        { date: "2024-03-02", description: "Kortköp", amount: -100 },
        {
          date: "2024-03-01",
          description: "APL*APPLE SEARCH ADS,800-275-2273,US",
          amount: -282.88,
        },
        { date: "2024-03-01", description: "Betalning", amount: -25000 },
        {
          date: "2024-03-01",
          description: "Avgift Utlandsbetalning",
          amount: -30,
        },
      ],
      raw: csv2,
    })
  ).toMatchSnapshot();
});
