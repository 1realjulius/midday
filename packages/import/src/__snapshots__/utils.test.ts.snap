// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Parse from comma delimiter 1`] = `
[
  [
    "Bokförd",
    "Valutadatum",
    "Text",
    "Typ",
    "Insättningar/uttag",
    "Bokfört",
    "saldo",
  ],
  [
    "2024-03-08",
    "2024-03-08",
    "ELLEVIO",
    "AB",
    "(PUBL)",
    "Betalning",
    "(bg/pg)",
    "-168",
    "00",
    "1942493",
    "52",
  ],
  [
    "2024-03-08",
    "2024-03-08",
    "ELLEVIO",
    "AB",
    "(PUBL)",
    "Betalning",
    "(bg/pg)",
    "-165",
    "00",
    "1942661",
    "52",
  ],
  [
    "2024-03-06",
    "2024-03-06",
    "BANKTJÄNSTER",
    "Annan",
    "-136",
    "00",
    "1942826",
    "52",
  ],
  [
    "2024-03-04",
    "2024-03-04",
    "KOSTNAD",
    "Annan",
    "-31",
    "85",
    "1942962",
    "52",
  ],
  [
    "2024-03-04",
    "2024-03-04",
    "H04510236587",
    "Internationell",
    "betalning",
    "230400",
    "00",
    "1942994",
    "37",
  ],
  [
    "2024-03-04",
    "2024-03-03",
    "AWS",
    "AMAZON",
    "C/24-03-02",
    "Kortköp",
    "-2797",
    "17",
    "1712594",
    "37",
  ],
  [
    "2024-03-04",
    "2024-03-02",
    "GOOGLE",
    "",
    "YOUT/24-03-01",
    "Kortköp",
    "-119",
    "00",
    "1715391",
    "54",
  ],
  [
    "2024-03-04",
    "2024-03-02",
    "DUBLIN",
    "",
    "",
    "",
    "",
    "",
    "/24-03-01",
    "Kortköp",
    "-119",
    "40",
    "1715510",
    "54",
  ],
  [
    "2024-03-04",
    "2024-03-02",
    "DUBLIN",
    "",
    "",
    "",
    "",
    "",
    "/24-03-01",
    "Kortköp",
    "-179",
    "11",
    "1715629",
    "94",
  ],
  [
    "2024-03-04",
    "2024-03-02",
    "GSUITE",
    "WDLIN/24-03-01",
    "Kortköp",
    "-29",
    "00",
    "1715809",
    "05",
  ],
  [
    "2024-03-04",
    "2024-03-02",
    "GOOGLE",
    "CLOUD/24-03-01",
    "Kortköp",
    "-39",
    "85",
    "1715838",
    "05",
  ],
  [
    "2024-03-01",
    "2024-03-01",
    "AMSTERDAM",
    "",
    "",
    "/24-03-01",
    "Kortköp",
    "-153",
    "10",
    "1715877",
    "90",
  ],
]
`;

exports[`Parse from colon delimiter 1`] = `
[
  [
    "Kontonummer",
    "Kontonamn",
    "",
    "Saldo",
    "Tillgängligt belopp",
  ],
  [
    "90248745117",
    "Rörelsekonto",
    "",
    "195 460,31",
    "195 460,31",
  ],
  [
    "Bokföringsdatum",
    "Transaktionsdatum",
    "Transaktionstyp",
    "Meddelande",
    "Belopp",
  ],
  [
    "2024-03-08",
    "2024-03-08",
    "Kortköp",
    "APL*APPLE SEARCH ADS,800-275-2273,US",
    "-387,27",
  ],
  [
    "2024-03-08",
    "2024-03-08",
    "Betalning",
    "SKATTEVERKET",
    "-44 134,00",
  ],
  [
    "2024-03-08",
    "2024-03-08",
    "Överföring",
    "TILL SKV",
    "44 134,00",
  ],
  [
    "2024-03-08",
    "2024-03-08",
    "Betalning",
    "hallon / Billogram",
    "-485,00",
  ],
  [
    "2024-03-07",
    "2024-03-06",
    "Kortköp",
    "WEBHALLENE",
    "-8 007,00",
  ],
  [
    "2024-03-07",
    "2024-03-07",
    "Utlandsbetalning",
    "ANK Utland Normal",
    "2 590,96",
  ],
  [
    "2024-03-06",
    "2024-03-06",
    "Överföring",
    "Skatt moms pwm",
    "-48 000,00",
  ],
  [
    "2024-03-06",
    "2024-03-06",
    "BG-insättning",
    "52762176",
    "137 500,00",
  ],
  [
    "2024-03-06",
    "2024-03-06",
    "Utlandsbetalning",
    "ANK Utland Normal",
    "1 923,95",
  ],
  [
    "2024-03-05",
    "2024-03-05",
    "Kortköp",
    "FIGMA MONTHLY RENEWAL,SAN FRANCISCO,US",
    "-316,17",
  ],
  [
    "2024-03-05",
    "2024-03-04",
    "Kortköp",
    "PAYPAL *APPERS GMBH,35314369001,AT",
    "-806,18",
  ],
  [
    "2024-03-02",
    "2024-03-01",
    "Kortköp",
    "GOOGLE*CLOUD 5NHT7Z,CC GOOGLE.COM,IE",
    "-1,22",
  ],
  [
    "2024-03-02",
    "2024-03-01",
    "Kortköp",
    "GSUITE,Dublin,IE",
    "-100,00",
  ],
]
`;

exports[`Parse from space delimiter 1`] = `
[
  [
    "Bokförd",
    "Valutadatum",
    "Text",
    "Typ",
    "Insättningar/uttag",
    "Bokfört saldo",
  ],
  [
    "2024-05-17",
    "2024-05-17",
    "SAN FRANCISC/24-05-16",
    "Kortköp",
    "-714",
    "26",
    "70001",
    "16",
  ],
  [
    "2024-05-15",
    "2024-05-15",
    "SKATTEVERKET",
    "Betalning (bg/pg)",
    "-67749",
    "00",
    "70715",
    "42",
  ],
  [
    "2024-05-08",
    "2024-05-08",
    "VOLKSWAGEN FINANS SVERIG",
    "Betalning (bg/pg)",
    "-6615",
    "00",
    "138464",
    "42",
  ],
  [
    "2024-05-07",
    "2024-05-07",
    "PREMIE FÖRS.",
    "Annan",
    "-5000",
    "00",
    "145079",
    "42",
  ],
  [
    "2024-05-06",
    "2024-05-06",
    "TWITTER PAID/24-05-05",
    "Kortköp",
    "-166",
    "00",
    "150079",
    "42",
  ],
  [
    "2024-05-06",
    "2024-05-06",
    "BANKTJÄNSTER",
    "Annan",
    "-264",
    "25",
    "150245",
    "42",
  ],
]
`;

exports[`Parse from dot delimiter 1`] = `
[
  [
    "Bokförd",
    "Valutadatum",
    "Text",
    "Typ",
    "Insättningar/uttag",
    "Bokfört",
    "saldo",
  ],
  [
    "2024-05-17",
    "2024-05-17",
    "SAN",
    "FRANCISC/24-05-16",
    "Kortköp",
    "-714",
    "26",
    "70001",
    "16",
  ],
  [
    "2024-05-15",
    "2024-05-15",
    "SKATTEVERKET",
    "Betalning",
    "(bg/pg)",
    "-67749",
    "00",
    "70715",
    "42",
  ],
  [
    "2024-05-08",
    "2024-05-08",
    "VOLKSWAGEN",
    "FINANS",
    "SVERIG",
    "Betalning",
    "(bg/pg)",
    "-6615",
    "00",
    "138464",
    "42",
  ],
  [
    "2024-05-07",
    "2024-05-07",
    "PREMIE FÖRS.",
    "Annan",
    "-5000",
    "00",
    "145079",
    "42",
  ],
  [
    "2024-05-06",
    "2024-05-06",
    "TWITTER",
    "PAID/24-05-05",
    "Kortköp",
    "-166",
    "00",
    "150079",
    "42",
  ],
  [
    "2024-05-06",
    "2024-05-06",
    "BANKTJÄNSTER",
    "Annan",
    "-264",
    "25",
    "150245",
    "42",
  ],
  [
    "2024-05-06",
    "2024-05-06",
    "KOSTNAD",
    "Annan",
    "-31",
    "85",
    "150509",
    "67",
  ],
  [
    "2024-05-06",
    "2024-05-06",
    "H06513143882",
    "Internationell",
    "betalning",
    "150000",
    "00",
    "150541",
    "52",
  ],
  [
    "2024-05-03",
    "2024-05-03",
    "DUBLIN",
    "",
    "",
    "",
    "",
    "",
    "/24-05-02",
    "Kortköp",
    "-99",
    "11",
    "541",
    "52",
  ],
]
`;

exports[`Transform transactions 1`] = `
[
  {
    "amount": 52,
    "date": "2024-03-08",
    "internal_id": "18b80bb7bd83fb53a3ebbab11e08923c83affae10b6f18e30eb50d2e981229e8",
    "manual": true,
    "name": "52",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": 52,
    "date": "2024-03-08",
    "internal_id": "18b80bb7bd83fb53a3ebbab11e08923c83affae10b6f18e30eb50d2e981229e8",
    "manual": true,
    "name": "52",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -119,
    "date": "2024-03-04",
    "internal_id": "3f14580ea53ce8e973bec152b9620d434a99719598afd33c2faf841d09c24866",
    "manual": true,
    "name": "54",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -179,
    "date": "2024-03-04",
    "internal_id": "08c2d0ca8316441ad0be7c2eb4a248ed95e540cc3833586e1d0e6781ebad3225",
    "manual": true,
    "name": "94",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": 90,
    "date": "2024-03-01",
    "internal_id": "a06c08077932d65ea5bc4b14ee2369764c5c2ea7275cb2c181c779b471bf7b46",
    "manual": true,
    "name": "90",
    "status": "posted",
    "team_id": "123",
  },
]
`;

exports[`Transform transactions 2`] = `
[
  {
    "amount": -387.27,
    "date": "2024-03-08",
    "internal_id": "03cfbf7fcb332caf7064f6f2467be135b1642506dab528a4cb0c1a0aca5e5024",
    "manual": true,
    "name": "Kortköp",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -44134,
    "date": "2024-03-08",
    "internal_id": "7e765db67d0d077fd5895ae000bc4460ad7280b8fa34c7ee56cdc1bad3bb9553",
    "manual": true,
    "name": "Betalning",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": 44134,
    "date": "2024-03-08",
    "internal_id": "767554ca99f7062d1bbce01147f0f1056447502945db3edc54f449156fb67021",
    "manual": true,
    "name": "Överföring",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -485,
    "date": "2024-03-08",
    "internal_id": "f320cf918ddd9ba96605990684a6874dbf3133932cac06f5c6941a3ce37ba0e2",
    "manual": true,
    "name": "Betalning",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -8007,
    "date": "2024-03-07",
    "internal_id": "d9bc8a82cb40a24902987a1cf58afb65739bb1c4cb8febc6dd2faf3403800c86",
    "manual": true,
    "name": "Kortköp",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": 2590.96,
    "date": "2024-03-07",
    "internal_id": "7f9cbac6ef9b5e82a66a428ec172b779c282a480eb69b08466bef6be3ddb2085",
    "manual": true,
    "name": "Utlandsbetalning",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -48000,
    "date": "2024-03-06",
    "internal_id": "4d79c96ec045889bf9d3b5883a36bbd3af7caede4225610837e7a00950b05da5",
    "manual": true,
    "name": "Överföring",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": 137500,
    "date": "2024-03-06",
    "internal_id": "942f9c9745f1fc95db08e74666259b92fa776b7f55d7c2b92b0c3cb7d97711dd",
    "manual": true,
    "name": "Bg Insättning",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": 1923.95,
    "date": "2024-03-06",
    "internal_id": "47ffb49f9e41a68c97baa7b7845c6c1b52739013e4dcc63113efcf7057304567",
    "manual": true,
    "name": "Utlandsbetalning",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -316.17,
    "date": "2024-03-05",
    "internal_id": "aa56f51414a8ce4cb41fa72343156031bf6411dc26ad634f7c494f25068e8630",
    "manual": true,
    "name": "Kortköp",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -806.18,
    "date": "2024-03-05",
    "internal_id": "8371c25e33ea7bf4b49be33867c03fd83db823690f066e11e3b63121cf5ffe1f",
    "manual": true,
    "name": "Kortköp",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -1.22,
    "date": "2024-03-02",
    "internal_id": "b46d4555ea35f6fe3f6cdbe5b5fdc64ec7ba5cf060f25d25b0559087621f605c",
    "manual": true,
    "name": "Kortköp",
    "status": "posted",
    "team_id": "123",
  },
  {
    "amount": -100,
    "date": "2024-03-02",
    "internal_id": "8434230c7374597561a0dd051a81e59fc7d757d85bfe29d6b2435275ea316c15",
    "manual": true,
    "name": "Kortköp",
    "status": "posted",
    "team_id": "123",
  },
]
`;
